core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
+      R__LOCKGUARD(gInterpreterMutex);
 
-   // If another thread executed LoadClassInfo at about the same time
-   // as this thread return early since the work was done.
-   if (!fCanLoadClassInfo) return;
+      // Return early if another thread loaded the info while we were waiting for the lock
+      if (fClassInfo)
+         return;
 
-   // If class info already loaded then do nothing.  This can happen if the
-   // class was registered by a dictionary, but the info came from reading
-   // the pch.
-   // Note: This check avoids using AutoParse for classes in the pch!
-   if (fClassInfo) {
-      return;
-   }
+      gInterpreter->AutoParse(GetName());
 
-   gInterpreter->AutoParse(GetName());
-   if (!fClassInfo) gInterpreter->SetClassInfo(const_cast<TClass*>(this));   // sets fClassInfo pointer
-   if (!gInterpreter->IsAutoParsingSuspended()) {
-      if (!fClassInfo) {
+      if (!fClassInfo)
+         gInterpreter->SetClassInfo(const_cast<TClass*>(this));
```

Comment:
because of the odd pattern and the static_cast being used, yes I felt compelled to do so (well in 3 out of the 4 actual calls ... but if somehow I am wrong that the pattern is odd/confusing, then indeed it is fine to omit the comment ...